fortran/src/H5_f.c

```diff
@@ -245,6 +245,14 @@ h5init_types_c(hid_t_f *types, hid_t_f *floatingtypes, hid_t_f *integertypes)
         return ret_value;
     if ((floatingtypes[3] = (hid_t_f)H5Tcopy(H5T_IEEE_F64LE)) < 0)
         return ret_value;
+    if ((floatingtypes[4] = (hid_t_f)H5Tcopy(H5T_IEEE_F16BE)) < 0)
```

Comment:
Were the IEEE 16 bit types missing before?

Reply:
Yes, I could add those in a separate PR if need be. They simply weren't added at the time support for _Float16 was introduced, but as this PR adds in a 16-bit floating point type it seemed reasonable to add them now.